farmdata2_modules/fd2_tabs/fd2_barn_kit/seedingReport.spec.js

```diff
@@ -56,6 +56,58 @@ describe('Testing for the seeding report page', () => {
 
         
     })
+    context.only('can see spinner at appropriate times', () => {
+        before(() => {
+            cy.login('manager1', 'farmdata2')
+            cy.visit('/farm/fd2-barn-kit/seedingReport')
+
+        })
+
+        it('show spinner after input', () => {
+            cy.get('[data-cy=start-date-select]')
+                .type('2019-01-01')
+            cy.get('[data-cy=end-date-select]')
+                .type('2019-03-01')
+            cy.get('[data-cy=generate-rpt-btn]').click()
+            cy.get('[data-cy=loader]').should('be.visible')
+        })
+
+        it('spinner dissappears after whole response returns 1 page', () => {
+            cy.get('[data-cy=start-date-select]')
+                .type('2019-01-01')
+            cy.get('[data-cy=end-date-select]')
+                .type('2019-02-01')
+            cy.get('[data-cy=generate-rpt-btn]').click()
+            cy.get('[data-cy=report-table]', { timeout: 20000 }).find('tr').its('length').then(length =>{
+                expect(length).to.equal(3)
+                cy.get('[data-cy=loader]').should('not.exist')
+            }) 
+        })
+        it.only('spinner remains before API return', () => {
+            cy.get('[data-cy=start-date-select]')
+                .type('2019-01-01')
+            cy.get('[data-cy=end-date-select]')
+                .type('2019-05-15')
+            cy.get('[data-cy=generate-rpt-btn]').click()
+            cy.get('[data-cy=report-table]', { timeout: 20000 }).find('tr').its('length').then(length =>{
+                expect(length).to.equal(101)
+                cy.get('[data-cy=loader]').should('exist')
+            })
+            cy.wait(10000)
+            cy.get('[data-cy=report-table]').find('tr').its('length').then(length =>{
```

Comment:
Would setting a timeout here work instead of using the wait? I think it would be less brittle with respect to running on different machines.

Reply:
I have not been able to use it. The problem is that with a 2 page request cypress sends the request, gets the request back, finds the table, gets the second page of the request, finds the row number and then checks it. This means that I am actually checking the row number of the wrong number of requests. I have gotten around this somewhat by sending a 3 page request and checking the second page rather than the first so that I can check what I was checking before. There must be a better way, as I have noticed inconsistent results on my own machine, perhaps due to API calls taking different amounts of time each time.

Reply:
Okay.  Let's just drop this test.  We have one that checks that the spinner appears when the report is being generated and we have one that checks that it is gone when the results are displayed.  I think that will be sufficient and would flag the most likely issues if something chagnes.